lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1201,6 +1220,63 @@ LLVMToSPIRV::transSpcvCast(CallInst* CI, SPIRVBasicBlock *BB) {
 }
 
 SPIRVValue *
+LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II, SPIRVBasicBlock *BB) {
+  auto getMemoryAccess = [](MemIntrinsic *MI)->std::vector<SPIRVWord> {
+    std::vector<SPIRVWord> MemoryAccess(1, MemoryAccessMaskNone);
+    if (SPIRVWord AlignVal = MI->getAlignment()) {
+      MemoryAccess[0] |= MemoryAccessAlignedMask;
+      MemoryAccess.push_back(AlignVal);
+    }
+    if (MI->isVolatile())
+      MemoryAccess[0] |= MemoryAccessVolatileMask;
+    return MemoryAccess;
+  };
+
+  switch (II->getIntrinsicID()) {
+  case Intrinsic::fmuladd : {
+    // For llvm.fmuladd.* fusion is not guaranteed. If a fused multiply-add
+    // is required the corresponding llvm.fma.* intrinsic function should be
+    // used instead.
+    SPIRVType *Ty = transType(II->getType());
+    SPIRVValue *Mul = BM->addBinaryInst(OpFMul, Ty,
+                                        transValue(II->getArgOperand(0), BB),
+                                        transValue(II->getArgOperand(1), BB),
+                                        BB);
+    return BM->addBinaryInst(OpFAdd, Ty, Mul,
+                             transValue(II->getArgOperand(2), BB), BB);
+  }
+  case Intrinsic::memset : {
+    // If memset is used for zero-initialization, find a variable which is being
+    // initialized and store null constant of the same type to this variable
+    MemSetInst *MSI = cast<MemSetInst>(II);
+    ConstantInt *Val = dyn_cast<ConstantInt>(MSI->getValue());
+    if (Val && Val->isZero()) {

```

Comment:
This only works if the size and alignment of the pointee type matches the length and alignment argument of memset instruction. This is not always true, so need to add these two conditions.